jekyll/_cci2/server/installation/phase-3-execution-environments.adoc

```diff
@@ -0,0 +1,1208 @@
+
+image::realitycheck-pipeline.png[Screenshot showing the realitycheck project building in the CircleCI app]
+
+[#vm-service]
+== 3. VM service
+
+VM service configures VM and remote docker jobs. You can configure a number of options for VM service, such as scaling rules. VM service is unique to AWS and GCP installations because it specifically relies on features of these cloud providers.
+
+ifndef::env-gcp[]
+
+[#aws]
+=== AWS
+
+[#set-up-security-group]
+==== Set up security group
+
+. *Get the information needed to create security groups*
++
+The following command returns your VPC ID (`vpcId`) and CIDR Block (`serviceIpv4Cidr`) which you need throughout this section:
++
+[source,shell]
+----
+aws eks describe-cluster --name=<your-cluster-name>
+----
+
+. *Create a security group*
++
+Run the following commands to create a security group for VM service:
++
+[source,shell]
+----
+aws ec2 create-security-group --vpc-id "<VPC_ID>" --description "CircleCI VM Service security group" --group-name "circleci-vm-service-sg"
+----
++
+This outputs a GroupID to be used in the next steps:
++
+[source, json]
+{
+    "GroupId": "<VM_SECURITY_GROUP_ID>"
+}
+
+. *Apply security group Nomad*
++
+Use the security group you just created and CIDR block values to apply the security group to the following. This will allow the VM service in your cluster to communicate on port 22 with created EC2 instances.
++
+[source,shell]
+----
+aws ec2 authorize-security-group-ingress --group-id "<VM_SECURITY_GROUP_ID>" --protocol tcp --port 22 --cidr "<SERVICE_IPV4_CIDR>"
+----
++
+For each https://github.com/CircleCI-Public/server-terraform/blob/main/nomad-aws/variables.tf#L1-L11[subnet] used by the nomad clients, find it's subnet cidr block add two rules with the following commands.
++
+Find the subnet cidr block with this command
++
+[source,shell]
+----
+aws ec2 describe-subnets --subnet-ids=<NOMAD_SUBNET_ID>
+----
++
+[source,shell]
+----
+# add a security group allowing docker access from nomad clients, to VM instances
+aws ec2 authorize-security-group-ingress --group-id "<VM_SECURITY_GROUP_ID>" --protocol tcp --port 2376 --cidr "<SUBNET_IPV4_CIDR>"
+----
++
+[source,shell]
+----
+# add a security group allowing SSH access from nomad clients, to VM instances
+aws ec2 authorize-security-group-ingress --group-id "<VM_SECURITY_GROUP_ID>" --protocol tcp --port 22 --cidr "<SUBNET_IPV4_CIDR>"
+----
+
+. *Apply the security group for SSH (If using public IPs for machines)*
++
+If using public IPs for VM service instances, run the following command to apply the security group rules so users can SSH into their jobs:
++
+[source,shell]
+----
+aws ec2 authorize-security-group-ingress --group-id "<VM_SECURITY_GROUP_ID>" --protocol tcp --port 54782 --cidr "0.0.0.0/0"
```

Comment:
@circleci/server is this right? `0.0.0.0/0`

Reply:
@rosieyohannan Yes this is correct. This is for all end users to SSH into the machine job.